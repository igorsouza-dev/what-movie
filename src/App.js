import React from 'react';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import history from './services/history';
import Routes from './routes';
import GlobalStyle from './styles/global';
import store from './store';

export default function App() {
  return (
    <Provider store={store}>
      <Router history={history}>
        <GlobalStyle />
        <Routes />
      </Router>
    </Provider>
  );
}
