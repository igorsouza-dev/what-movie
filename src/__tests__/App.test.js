import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from 'store/reducers';
import App from 'App';
import apiMock from 'services/api';
import { getYear, startOfWeek, endOfWeek } from 'date-fns';

jest.mock('services/api');

function getDateOnly(date) {
  return date.toISOString().substring(0, 10);
}
const today = new Date();
const year = getYear(today);
const lastYear = year - 1;
const start = getDateOnly(startOfWeek(today));
const end = getDateOnly(endOfWeek(today));
const urls = [
  `/discover/movie?primary_release_date.gte=${start}&primary_release_date.lte=${end}&sort_by=vote_average.desc`,
  `/discover/movie?primary_release_year=${year}&sort_by=vote_average.desc&vote_count.gte=200`,
  `/discover/movie?primary_release_year=${lastYear}&sort_by=vote_average.desc&vote_count.gte=1000`,
];

function renderWithRedux(
  ui,
  { initialState, mockStore = createStore(reducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={mockStore}>{ui}</Provider>),

    mockStore,
  };
}

describe('<App />', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container, getByText } = renderWithRedux(<App />);
    const linkElement = getByText(/What movie?/i);

    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const { getAllByText } = renderWithRedux(<App />);
    apiMock.get.mockResolvedValueOnce({
      data: {
        results: [
          {
            id: '1234',
            original_title: 'Movie name test',
          },
        ],
      },
    });

    expect(getAllByText('Loading...')).toHaveLength(3);
    // expect(apiMock.get).toHaveBeenCalledTimes(3);
    expect(apiMock.get).toHaveBeenCalledWith(urls[0]);
    expect(apiMock.get).toHaveBeenCalledWith(urls[1]);
    expect(apiMock.get).toHaveBeenCalledWith(urls[2]);
  });
});
