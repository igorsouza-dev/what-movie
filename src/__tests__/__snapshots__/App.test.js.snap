// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<div
  class="sc-kGXeez ehUfZi"
>
  <h1
    class="sc-bxivhb fSvbbt"
    color="#d5dcf9"
    data-testid="logo"
    size="80"
  >
    What Movie?
  </h1>
  <form
    class="sc-bdVaJa hjizjZ"
    data-testid="search-bar"
  >
    <input
      class="sc-bwzfXH jEOSHL"
      placeholder="Type the name of a movie"
      type="text"
      value=""
    />
    <button
      class="sc-htpNat dLCsyj"
      type="submit"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </button>
  </form>
  <div
    class="sc-kgoBCf jANwOR"
  >
    <h2
      class="sc-chPdSV byfFmt"
    >
      Opening this week
    </h2>
    <div>
      Loading...
    </div>
    
  </div>
  <div
    class="sc-kgoBCf jANwOR"
  >
    <h2
      class="sc-chPdSV byfFmt"
    >
      Top movies of 2020
    </h2>
    <div>
      Loading...
    </div>
    
  </div>
  <div
    class="sc-kgoBCf jANwOR"
  >
    <h2
      class="sc-chPdSV byfFmt"
    >
      Top movies of last year
    </h2>
    <div>
      Loading...
    </div>
    
  </div>
</div>
`;

exports[`<App /> should render correctly 1`] = `
<div
  class="sc-kGXeez ehUfZi"
>
  <h1
    class="sc-bxivhb fSvbbt"
    color="#d5dcf9"
    data-testid="logo"
    size="80"
  >
    What Movie?
  </h1>
  <form
    class="sc-bdVaJa hjizjZ"
    data-testid="search-bar"
  >
    <input
      class="sc-bwzfXH jEOSHL"
      placeholder="Type the name of a movie"
      type="text"
      value=""
    />
    <button
      class="sc-htpNat dLCsyj"
      type="submit"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </button>
  </form>
  <div
    class="sc-kgoBCf jANwOR"
  >
    <h2
      class="sc-chPdSV byfFmt"
    >
      Opening this week
    </h2>
    <div
      class="sc-kAzzGY gPTTjj"
      type="flex"
    >
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 1"
            src="https://image.tmdb.org/t/p/w1851.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 1
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 2"
            src="https://image.tmdb.org/t/p/w1852.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 2
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 3"
            src="https://image.tmdb.org/t/p/w1853.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 3
            </strong>
          </div>
        </div>
      </div>
    </div>
    
  </div>
  <div
    class="sc-kgoBCf jANwOR"
  >
    <h2
      class="sc-chPdSV byfFmt"
    >
      Top movies of 2020
    </h2>
    <div
      class="sc-kAzzGY gPTTjj"
      type="flex"
    >
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 1"
            src="https://image.tmdb.org/t/p/w1851.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 1
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 2"
            src="https://image.tmdb.org/t/p/w1852.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 2
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 3"
            src="https://image.tmdb.org/t/p/w1853.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 3
            </strong>
          </div>
        </div>
      </div>
    </div>
    
  </div>
  <div
    class="sc-kgoBCf jANwOR"
  >
    <h2
      class="sc-chPdSV byfFmt"
    >
      Top movies of last year
    </h2>
    <div
      class="sc-kAzzGY gPTTjj"
      type="flex"
    >
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 1"
            src="https://image.tmdb.org/t/p/w1851.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 1
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 2"
            src="https://image.tmdb.org/t/p/w1852.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 2
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-htoDjs kKAWrx"
      >
        <div
          class="sc-iwsKbI gCoABE"
        >
          /10
        </div>
        <div
          class="sc-cSHVUG cIcywD"
          data-testid="poster-img-container"
        >
          <img
            alt="Movie name test 3"
            src="https://image.tmdb.org/t/p/w1853.jpg"
          />
        </div>
        <div
          class="sc-gZMcBi FECGe"
        >
          <div
            class="sc-dnqmqq kabniG"
            data-testid="name-container"
          >
            <strong>
              Movie name test 3
            </strong>
          </div>
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
